### DISCUSSION TITLE
Add serve-index as connect middleware for @parcel/reporter-dev-server

### DISCUSSION THREAD

**Discussion by kai-dorschner-twinsity** [parcel-bundler/parcel#7575] [Ideas]
**Use Case**
When loading parcel in dev mode with multiple entry points (`parcel serve ./src/*.html --open `) it would be nice to see a directory-listing and choose from there what file you want to see.

If there is more people than just me I can possibly produce a draft plugin / PR for the existing [dev server](https://github.com/parcel-bundler/parcel/blob/v2/packages/reporters/dev-server/src/Server.js).

**Vision**
It could also be useful to have an API to add _any_ middleware to the dev server, not just this in particular.

Ref: https://www.npmjs.com/package/serve-index

**Comment by diraneyya** [comment]
I am interested in one of these two things:
- Either that the `parcel [serve]` command can serve a folder index page  for certain directories
- Or that Node's "readdirSync" be emulated in a manner similar to "readfileSync"

Whichever is more convenient and inline with the philosophy of the project.
